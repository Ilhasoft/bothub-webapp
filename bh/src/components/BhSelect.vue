<template>
  <div class="bh-select">
    <select
      :disabled="disabled"
      :class="{
        'bh-select': true,
        'bh-select__disabled': disabled,
      }"
    >
      <slot />
    </select>
  </div>
</template>

<script>
import BhText from './BhText';


export default {
  name: 'BhSelect',
  extends: BhText,
  props: {
    disabled: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      selected: null,
    };
  },
};
</script>

<style lang="scss">
@import '@scss/colors.scss';
@import '@scss/variables.scss';
@import '@scss/forms.scss';

.bh {
  &-select {
    @include text();

    width: 100%;

    &__disabled {
      cursor: no-drop;
    }
  }

  &-select:focus {
    outline: 2px solid $color-primary;
  }
}
</style>
