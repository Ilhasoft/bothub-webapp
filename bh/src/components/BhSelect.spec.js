/* eslint-env jest */
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { compileToFunctions } from 'vue-template-compiler';

import BH from '..';
import BhSelect from './BhSelect';


const localVue = createLocalVue();
localVue.use(BH);

const defaultSlotOptionsComponent = compileToFunctions('<option value="Milk">Milk</option>');

describe('BhSelect.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(BhSelect, {
      localVue,
      slots: {
        default: defaultSlotOptionsComponent,
      },
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('BhSelect Disabled renders correctly', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(BhSelect, {
      localVue,
      slots: {
        default: defaultSlotOptionsComponent,
      },
    });
    wrapper.setProps({
      disabled: true,
    });
  });

  test('Disabled Select input', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
