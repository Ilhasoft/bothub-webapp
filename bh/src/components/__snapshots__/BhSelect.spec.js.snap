// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BhSelect.vue BhSelect Disabled renders correctly Disabled Select input 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;

exports[`BhSelect.vue renders correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
